libs/langchain/langchain/llms/huggingface_text_gen_inference.py

```diff
@@ -193,39 +177,90 @@ async def _acall(
         run_manager: Optional[AsyncCallbackManagerForLLMRun] = None,
         **kwargs: Any,
     ) -> str:
+        if self.streaming:
+            completion = ""
+            async for chunk in self._astream(prompt, stop, run_manager, **kwargs):
+                completion += chunk.text
+            return completion
+
         invocation_params = self._invocation_params(stop, **kwargs)
-        if not self.stream:
-            res = await self.async_client.generate(
-                prompt,
-                **invocation_params,
-            )
-            # remove stop sequences from the end of the generated text
+        res = await self.async_client.generate(prompt, **invocation_params)
+        # remove stop sequences from the end of the generated text
+        for stop_seq in invocation_params["stop_sequences"]:
+            if stop_seq in res.generated_text:
+                res.generated_text = res.generated_text[
+                    : res.generated_text.index(stop_seq)
+                ]
+        return res.generated_text
+
+    def _stream(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> Iterator[GenerationChunk]:
+        invocation_params = self._invocation_params(stop, **kwargs)
+
+        for res in self.client.generate_stream(prompt, **invocation_params):
+            # identify stop sequence in generated text, if any
```

Comment:
The stream logic for stop sequence didn't match the batch one, so took opp to fix